### ISSUE TITLE
Reversed dimension values on different devices/emulators

### ISSUE THREAD

**Issue by DDeveloper95** [flutter/flutter#25286]
Hello Flutter Team,
I'm decoding an image as shown below:
`File picture = File(path);`
`Image image = decodeImage(picture.readAsBytesSync());`
I get the file through the Flutter Image_Picker Plugin.
The picture was taken in portrait orientation.

But I'm getting different dimension results with the same picture on different Android OS.
### Oreo 8.1
![image](https://user-images.githubusercontent.com/32966133/49886454-f877ec80-fe39-11e8-8509-902cbd7b800e.png)

### Nougat 9.0
![image](https://user-images.githubusercontent.com/32966133/49887781-7f7a9400-fe3d-11e8-81c6-4df57710e2f1.png)

### Nougat 7.0.0
On this Version the Google Photo App even shows reversed values in width and height.
![image](https://user-images.githubusercontent.com/32966133/49887544-f19ea900-fe3c-11e8-8c66-cc2d81be6e43.png)

Therefore I'm really confused and I'm not sure where the problem is since the same code is running on all emulators. I also noticed that if I debug the code on a normal device the (Samsung Galaxy S6 and Google Pixel 3XL) the values are as shown below: 
![image](https://user-images.githubusercontent.com/32966133/49888383-27dd2800-fe3f-11e8-9e17-d531d8953e32.png)

## Steps to Reproduce

1.  Create different Emulators and run code.

**Comment by zoechi** [comment]
Please add the output of `flutter doctor -v`.

It is not clear to me if you are reporting an issue with decoding or with the camera_plugin.

There are several issues about the camera_plugin already
#25232, #19458, #17606, #16587

**Comment by DDeveloper95** [comment]
Here is the flutter doc output:
```
[√] Flutter (Channel beta, v1.0.0, on Microsoft Windows [Version 10.0.17763.134], locale de-DE)
    • Flutter version 1.0.0 at C:\Users\didde\flutter
    • Framework revision 5391447fae (13 days ago), 2018-11-29 19:41:26 -0800
    • Engine revision 7375a0f414
    • Dart version 2.1.0 (build 2.1.0-dev.9.4 f9ebf21297)

[√] Android toolchain - develop for Android devices (Android SDK 28.0.3)
    • Android SDK at C:\Users\didde\AppData\Local\Android\Sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)
    • All Android licenses accepted.

[√] Android Studio (version 3.2)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 31.1.1
    • Dart plugin version 181.5656
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)

[!] IntelliJ IDEA Ultimate Edition (version 2018.2)
    • IntelliJ at C:\Users\didde\AppData\Local\JetBrains\Toolbox\apps\IDEA-U\ch-0\182.4505.22
    X Flutter plugin not installed; this adds Flutter specific functionality.
    X Dart plugin not installed; this adds Dart specific functionality.
    • For information about installing plugins, see
      https://flutter.io/intellij-setup/#installing-the-plugins

[√] Connected device (3 available)
    • Android SDK built for x86 • emulator-5554 • android-x86 • Android 8.1.0 (API 27) (emulator)
    • Android SDK built for x86 • emulator-5556 • android-x86 • Android 7.0 (API 24) (emulator)
    • Android SDK built for x86 • emulator-5558 • android-x86 • Android 9 (API 28) (emulator)

! Doctor found issues in 1 category.
```
I've tested some more and it doesn't matter if I take the image with the camera plugin or the image_picker plugin the height and width values are always reversed based on the device or emulator I'm using after the decoder decoded the image.
I've found this Issue #10895 which is refering only one part of my Problem. Although, even if the values are reversed the image is always shown in the right direction in the Image Widget.

But I would rather guess that the problem is causing decoder and not the plugins.

**Comment by DDeveloper95** [comment]
Maybe I found where the problem is:
It seems like the decoder reverses the width and height based on the exif orientation
Here is a [explanation](https://www.impulseadventure.com/photo/exif-orientation.html)

**Comment by zoechi** [comment]
> the problem is causing decoder and not the plugins.

It should be possible to find out by using the same image file on different devices (for example by adding it as resource and load and decode it from there)

**Comment by DDeveloper95** [comment]
I'm still confused why I get different results on different devices and it seems like the decoder doesn't always get the orientation out of the exif although it should be provided. I tested it on an Samsung Galaxy S6 where I get the orientation but on my Google Pixel 3 XL it won't extract the orientation. Therefore the decoder always reverses the height and width on my Pixel since no orientation is provided.

**Comment by DDeveloper95** [comment]
> > the problem is causing decoder and not the plugins.
> 
> It should be possible to find out by using the same image file on different devices (for example by adding it as resource and load and decode it from there)

I will try this. I'll keep updated.

**Comment by DDeveloper95** [comment]
So me and my partner made some more tests and it sems like really the decoder is the problem. It doesn't extract the exif information on some devices (Pixel 3 XL and Nexus 9). We also used always the same picture (it was taken with the nexus 9).

We came to the conclusion that only images taken by Samsung devices have a correct decoded exif information and on the most other devices the information won't be decoded. Therefore the decoder works only on Samsung devices eventhough Samsungs are not behaving correctly storing this information [look here](https://stackoverflow.com/questions/39400351/android-exif-data-always-0-how-to-change-it).

We found a workaround when we used the [dart package exif](https://pub.dartlang.org/packages/exif) by extracting the image information.

This is the code we used.
```
Map<String, dynamic> exif = await readExifFromBytes(widget.imageFile.readAsBytesSync());

    String width = exif['EXIF ExifImageWidth'].toString();
    String height = exif['EXIF ExifImageLength'].toString();

    if(exif['Image Orientation'].toString() == 'Rotated 90 CW' ||
        exif['Image Orientation'].toString() == 'Rotated 90 CCW'){
      return Size(double.parse(height), double.parse(width));
    }else{
      return Size(double.parse(width), double.parse(height));
    }
```
